src/v/redpanda/admin_server.cc

```diff
@@ -3764,6 +3770,26 @@ ss::future<> admin_server::restart_redpanda_service(service_kind service) {
               "Unknown issue restarting schema_registry");
         }
     }
+
+    if (service == service_kind::http_proxy) {
+        // Checks specific to http proxy
+        if (_http_proxy == nullptr) {
+            throw ss::httpd::server_error_exception(
+              "Pandaproxy is undefined. Is it set in the .yaml config "
+              "file?");
+        }
+
+        try {
+            co_await _http_proxy->restart();
+        } catch (const std::exception& ex) {
+            vlog(
+              logger.error,
+              "Unknown issue restarting http_proxy: {}",
+              ex.what());
+            throw ss::httpd::server_error_exception(
+              "Unknown issue restarting http_proxy");
+        }
+    }
```

Comment:
This is copy-paste code from restarting the schema registry. Please, refactor into a method.